engine/src/main/java/org/terasology/rendering/opengl/FBO.java

```diff
@@ -100,12 +116,69 @@ public void unbind() {
         glBindFramebufferEXT(GL_FRAMEBUFFER_EXT, 0);
     }
 
+
+
+    /**
+     *  Get the texture ID for the identifier
+     * @param identifier
+     * @return
+     */
+    public static ResourceUrn getTextureUrn(Dimensions dimensions, FBO fboId, String identifier) {
+        return TextureUtil.getTextureUriForEmpty(dimensions.width(),dimensions.height(),"fbo_"+fboId.fboId+"_" + identifier);
+    }
+
+    /**
+     * returns an existing texture in Assets or creates a new instance if one does not exist
+     * @param identifier
+     * @return
+     */
+    //TODO: need a method to bind the texture to the FBO
+    private Texture getTextureInstance(String identifier) {
+        ResourceUrn urn = FBO.getTextureUrn(dimensions,this,identifier);
+
+        Texture texture = Assets.getTexture(urn).get();
+        this.textureMapping.put(urn.toString(),texture);
+        return texture;
+    }
+
+    /**
+     *   returns an existing texture in Assets or creates a new instance if one does not exist
+     * @param buffer
+     * @return
+     */
+    //TODO: needs a method to bind the texture to the FBO
+    private Texture getTextureInstance(BaseFboBuffer buffer) {
+        return  this.getTextureBuffer(buffer.getIdentifier());
+    }
+
+    /**
+     * returns a texture object for the identifier else returns null if one does not exist
+     * @param identifier
+     * @return a texture bound to fhte FBO
+     */
+    public Texture getTextureBuffer(String identifier) {
+        ResourceUrn resource = this.getTextureUrn(dimensions,this,identifier);
+        if(this.textureMapping.containsKey(resource.toString()))
+            return null;
+        return this.textureMapping.get(resource.toString());
+    }
+
+    /**
+     * get a texture Instance
+     * @param buffer
+     * @return
+     */
+    public Texture getTextureBuffer(BaseFboBuffer buffer) {
+        return  this.getTextureBuffer(buffer.getIdentifier());
+    }
+
     /**
      * Binds the color attachment to the currently active texture unit.
      * Once a texture is bound it can be sampled by shaders.
      */
     public void bindTexture() {
-        GL11.glBindTexture(GL11.GL_TEXTURE_2D, colorBufferTextureId);
+        Texture texture = this.getTextureBuffer(BaseFboBuffer.COLOR_TEXTURE);
+        GL11.glBindTexture(GL11.GL_TEXTURE_2D, texture.getId());
```

Comment:
I'll refrain from offering feedback on the overall approach to let you develop it and see where we end up.

However, please be aware that at least some (probably all) of these binding methods are used multiple times per frame. As such they shouldn't contain any instantiation nor should they call methods where anything gets instantiated.